chef/cookbooks/nova_dashboard/recipes/server.rb

```diff
@@ -17,6 +17,30 @@
 include_recipe "apache2::mod_wsgi"
 include_recipe "apache2::mod_rewrite"
 
+keystone_settings = KeystoneHelper.keystone_settings(node, @cookbook_name)
+
+# Multidomain can only be enabled if the API version of Keystone is
+# equal or bigger that v3.
+keystone_api_version = keystone_settings['api_version']
+keystone_internal_auth_url = keystone_settings['internal_auth_url']
+
+# TODO(aplanas): Remove this line when keystone API 'v3' works for
+# every OpenStack configuration file.
+if node["nova_dashboard"]["multidomain"] && keystone_api_version.to_f < 3
+  keystone_api_version = '3'
+  keystone_internal_auth_url = KeystoneHelper.versioned_service_URL(
+    nil, keystone_settings["internal_url_host"],
+    keystone_settings["service_port"], keystone_api_version, node,
+    @cookbook_name)
+end
+
+# TODO(aplanas): Uncomment this when keystone API 'v3' works for every
+# OpenStack configuration file.
+# if node["nova_dashboard"]["multidomain"] && keystone_api_version.to_f < 3
+#     Chef::Application.fatal!("Multidomain feature can't be deployed for Keystone API ${keystone_api_version}")
+# end
```

Comment:
To be honest, I'm not sure it's worth keeping that as when everything will be v3-ready, we will likely not allow v2.0 by default. But I don't mind keeping that anyway.